packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -96,12 +102,20 @@ export class DeviceSession implements Disposable {
   }
 
   public async perform(type: ReloadAction) {
+    const clearDebugContext = async () => {
+      // clears the previous debug session to avoid any conflicts
+      await this.debugSession.stopDebugSession();
+      await this.debugSession.startDebugSession();
```

Comment:
Similatily here, I understand that we are referring to JS debugger here. We don't need to leak the implementation details here. If we want to have a signal that the session needs to be cleared, we should expose a mathod for that and the session on its own will decide that it should restart the JS debugger connection.